pkg/api/defaults.go

```diff
@@ -444,23 +444,25 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 			}
 
 			// Distro assignment for masterProfile
-			if cs.Properties.MasterProfile.Distro == "" && cs.Properties.MasterProfile.ImageRef == nil {
-				if cs.Properties.OrchestratorProfile.IsKubernetes() && cs.Properties.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage == "" {
-					cs.Properties.MasterProfile.Distro = AKSUbuntu1604
-				} else {
-					cs.Properties.MasterProfile.Distro = Ubuntu
+			if cs.Properties.MasterProfile.ImageRef == nil {
```

Comment:
I believe previous if-then block was a bug (fall-through to else if wrongly), and this is a correct, desired logic (i.e. only overwrite default Distro when ImageRef is not specified).